pkg/gofr/datasource/surrealdb/surrealdb.go

```diff
@@ -0,0 +1,543 @@
+package surrealdb
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"net/url"
+	"os"
+
+	"github.com/surrealdb/surrealdb.go"
+	"github.com/surrealdb/surrealdb.go/pkg/connection"
+	"github.com/surrealdb/surrealdb.go/pkg/constants"
+	"github.com/surrealdb/surrealdb.go/pkg/logger"
+	"github.com/surrealdb/surrealdb.go/pkg/models"
+
+	"go.opentelemetry.io/otel/trace"
+)
+
+var (
+	// errNotConnected indicates that the database client is not connected.
+	errNotConnected = errors.New("not connected to database")
+)
+
+const (
+	https = "https"
+	ws    = "ws"
```

Comment:
Please use this

```suggestion
	schemeHTTPS = "https"
	schemeWS    = "ws"
```

Otherwise the constants names are meaningless